README.md

```diff
@@ -19,81 +31,90 @@ The build script creates two networks, `leftnet` (192.168.0.0/24) and `rightnet`
 ```bash
 docker network create leftnet --subnet 192.168.0.0/24
 docker network create rightnet --subnet 192.168.100.0/24
 ```
 
-The networks can be removed by executing
+The networks can be removed by executing:
+
 ```bash
 docker network rm leftnet rightnet
 ```
 
-### Running the simulator
+## Running the simulator
 
 To run the simulator, you need to set up the networks first, as described above.
 After that, build and run the simulator:
+
 ```bash
 ./run.sh "simple-p2p --delay=10ms --bandwidth=10Mbps"
 ```
 
-All paramters to `run.sh` are passed to waf, i.e. the command run inside the container will be `./waf --run "simple-p2p --delay=10ms --bandwidth=10Mbps"`.
+All paramters to `run.sh` are passed to waf, i.e. the command run inside the
+container will be `./waf --run "simple-p2p --delay=10ms --bandwidth=10Mbps"`.
 
-### Building your own QUIC endpoint
+## Building your own QUIC endpoint
 
-The [endpoint](endpoint) directory contains the base docker image for an endpoint container. 
-The pre-built image is available on [dockerhub](https://hub.docker.com/r/martenseemann/quic-network-simulator-endpoint).
+The [endpoint](endpoint) directory contains the base docker image for an
+endpoint container.  The pre-built image is available on
+[dockerhub](https://hub.docker.com/r/martenseemann/quic-network-simulator-endpoint).
 
-When building a Docker image for your own QUIC implementation, there are two files you need to create: `Dockerfile` and `run_endpoint.sh`.
-Copy this Dockerfile and add the commands to build your QUIC implementation.
-```docker
-FROM martenseemann/quic-network-simulator-endpoint:latest
+Follow these steps to build a Docker image for your own QUIC implementation:
 
-# download and build your QUIC implementation
-# [ DO WORK HERE ]
+1. Create a new directory for your implementation. You will create two files in
+   this directory: `Dockerfile` and `run_endpoint.sh`, as described below.
 
-# copy run script and run it
-COPY run_endpoint.sh .
-RUN chmod +x run_endpoint.sh
-ENTRYPOINT [ "./run_endpoint.sh" ]
-```
+1.  Copy the Dockerfile below and add the commands to build your QUIC
+    implementation.
 
-`run_endpoint.sh` could look like this:
-```bash
-# Set up the routing needed for the simulation.
-/setup.sh
+    ```dockerfile
+    FROM martenseemann/quic-network-simulator-endpoint:latest
 
-ROLE=$1
-shift
+    # download and build your QUIC implementation
+    # [ DO WORK HERE ]
 
-if [ "$ROLE" == "client" ]; then
-    [ INSERT COMMAND TO RUN YOUR QUIC CLIENT ]
-elif [ "$ROLE" == "server" ]; then
-    [ INSERT COMMAND TO RUN YOUR QUIC SERVER ]
-fi
-```
+    # copy run script and run it
+    COPY run_endpoint.sh .
+    RUN chmod +x run_endpoint.sh
+    ENTRYPOINT [ "./run_endpoint.sh" ]
+    ```
 
-Build your image and assign a tag (`$YOURTAG`):
-```
-docker build . -t $YOURTAG
-```
+    1. Now, copy the script below into `run_endpoint.sh`, and add commands as instructed.
```

Comment:
For some reason this shows up as "i." in the rendered Markdown. Maybe the indentation is wrong?